accepted/2024/supplied-by-platform.md

```diff
@@ -0,0 +1,192 @@
+
+Even .NET 9, there are certain assemblies that ship as both packages, but are also part of certain shared frameworks such as ASP.NET. See details <https://github.com/dotnet/aspnetcore/issues/3609>. An example of such assembly/package is `Microsoft.Extensions.Logging`, which is [published on nuget.org](https://www.nuget.org/packages/Microsoft.Extensions.Logging/), but also part of the Microsoft.AspNetCore.App shared framework. Currently there's conflict resolution in the .NET SDK to ensure that the latest version is chosen.
+
+There are a few benefits:
+
+It is trivial to make the assumption that the fewer packages need to be downloaded, the better the performance will be.
+Beyond that, the extra packages within the graph, do make the resolution step more challenging. Some of the targeting packs used to bring in such a large package graph, that affected the resolution performance significantly. See  and <https://github.com/NuGet/Home/issues/11993> for more details. Furthermore, certain versions of popular packages have taken dependencies on these targeting packs for historical reasons. Such an example is <https://www.nuget.org/packages/log4net/2.0.10> which performs significantly better than <https://www.nuget.org/packages/log4net/2.0.9> when installed in a .NET (Core) projects, see <https://github.com/NuGet/Home/issues/10030>.
+
+Certain versions of these packages are no longer  part of the project graph, thus reducing the chances of false positive scanning.
+This is especially important in the context of assemblies that are part of shared frameworks. The build time conflict resolution ensures that the most up to date version is used, despite the fact that the user referenced package may habe vulnerabilities.
+Examples: <https://github.com/dotnet/sdk/issues/30659#issuecomment-2072567192>.
+
+This changes significantly helps align what gets restored and what gets published. By avoiding packages that are simply not going to be used at runtime, what gets used at runtime becomes more transparent.
+
+## Explanation
+
+### Functional explanation
+
+<!-- Explain the proposal as if it were already implemented and you're teaching it to another person. -->
+<!-- Introduce new concepts, functional designs with real life examples, and low-fidelity mockups or  pseudocode to show how this proposal would look. -->
+
+A list of package id/versions to be pruned will be provided by the .NET SDK. The version will signify the highest version to be pruned.
+When NuGet encounters any of the specified packages or lower, it will simply remove the package from the graph. This package id will not appear in the assets file, but there will be a detailed verbosity message indicating that the package id has been pruned for the given framework.
+
+The feature is framework specific, and can be opted in/out using the `NuGetPrunePlatformPackages` property.
+In .NET 10 SDK, this feature will be enabled for all .NET (Core) target frameworks newer than netcoreapp3.0 by default.
```

Comment:
Wouldn't .NET Framework benefit from this as well i.e. when referencing .NET Standard 1.x packages?

Reply:
Potentially, yeah. 
I can add it.